src/routes/contact.js

```diff
@@ -191,12 +191,71 @@ router.post('/send', async (req, res) => {
       email.sendContactCheckInEmail(e)
     })
 
-    response.setMessage('Contacts emailed')
+    response.setMessage({
+      results: {
+        message: 'Contacts emailed',
+        total: emails.length
+      }
+    })
+  } catch (e) {
+    console.error(e)
+    response.setCode(500)
+  }
+
+  return res.status(response.getCode()).send(response.getMessage())
+})
+
+
+router.post('/send/:type/:id', async (req, res) => {
+  const response = new utils.Response()
+
+  try {
+    let entity
+    if (req.params.type.toLowerCase() === 'entity') {
+      entity = await models.Entity.findById(req.params.id)
+    } else if (req.params.type.toLowerCase() === 'contact') {
+      entity = await models.Contact.findOne({
+        where: {
+          id: req.params.id
+        }
+      })
+    }
+
+    if (entity.email !== null) {
+      const primary = entity.email.filter(e => e.isPrimary === 'true').length ? entity.email.filter(e => e.isPrimary === 'true')[0] : entity.email[0]
+      // short-lived temporary token that only lasts one hour
+      const temporaryToken = await utils.getToken(req.params.id, primary.address, 'Entity')
+
+      const e = {
+        email: primary.address,
+        name: entity.name,
+        entityName: entity.name,
+        entityId: entity.id,
+        token: temporaryToken
+      }
+      email.sendContactCheckInEmail(e).then(() => {
+        response.setMessage(`${entity.name} emailed sent.`)
+        response.setCode(200)
+      }, err => {
+        response.setMessage('There was an error: ' + err)
+        response.setCode(500)
+      })
+    } else {
+      response.setMessage('Email Address not found.')
+      response.setCode(500)
```

Comment:
Is a server error appropriate here? This is a data problem with the specific user.

Reply:
Do you have a different status code in mind? Does "422 Unprocessable Entity"  make sense?

https://stackoverflow.com/questions/3050518/what-http-status-response-code-should-i-use-if-the-request-is-missing-a-required#:~:text=The%20422%20(Unprocessable%20Entity)%20status,was%20unable%20to%20process%20the

Reply:
@mschechter-bellese ^^

Reply:
422 is definitely something we need to move away from; it's specific to WebDAV and should be only used in those types of cases. 500 is inappropriate because there was no server error at all.

In this case, there's nothing wrong with the provided entity (the request), and the user exists; they just don't have an e-mail address. It's not really clear what this should be because the field is optional. The closest one I can find is that because the resource has missing data we return a 404 with an explanation. This is a short-term solution, as the new API (I think) avoids this scenario entirely.

Reply:
Thanks for the feedback @mschechter-bellese .. I've changed this status code to 404.